examples/msaa/main.rs

```diff
@@ -0,0 +1,181 @@
+        }),
+        rasterization_state: wgpu::RasterizationStateDescriptor {
+            front_face: wgpu::FrontFace::Ccw,
+            cull_mode: wgpu::CullMode::None,
+            depth_bias: 0,
+            depth_bias_slope_scale: 0.0,
+            depth_bias_clamp: 0.0,
+        },
+        primitive_topology: wgpu::PrimitiveTopology::TriangleList,
+        color_states: &[wgpu::ColorStateDescriptor {
+            format: wgpu::TextureFormat::Bgra8Unorm,
+            color_blend: wgpu::BlendDescriptor::REPLACE,
+            alpha_blend: wgpu::BlendDescriptor::REPLACE,
+            write_mask: wgpu::ColorWrite::ALL,
+        }],
+        depth_stencil_state: None,
+        index_format: wgpu::IndexFormat::Uint16,
+        vertex_buffers: &[],
+        sample_count,
+    });
+
+    let mut swap_chain = device.create_swap_chain(
+        &surface,
+        &wgpu::SwapChainDescriptor {
+            usage: wgpu::TextureUsage::OUTPUT_ATTACHMENT,
+            format: wgpu::TextureFormat::Bgra8Unorm,
+            width: size.width.round() as u32,
+            height: size.height.round() as u32,
+        },
+    );
+    let mut running = true;
+    while running {
+        events_loop.poll_events(|event| match event {
+            Event::WindowEvent { event, .. } => match event {
+                WindowEvent::KeyboardInput {
+                    input:
+                        KeyboardInput {
+                            virtual_keycode: Some(code),
+                            state: ElementState::Pressed,
+                            ..
+                        },
+                    ..
+                } => match code {
+                    VirtualKeyCode::Escape => running = false,
+                    _ => {}
+                },
+                WindowEvent::CloseRequested => running = false,
+                _ => {}
+            },
+            _ => {}
+        });
+
+        let frame = swap_chain.get_next_texture();
+        let multisampled_texture_extent = wgpu::Extent3d {
+            width: size.width.round() as u32,
+            height: size.height.round() as u32,
+            depth: 1
+        };
+        let multisampled_frame_descriptor = &wgpu::TextureDescriptor {
+            size: multisampled_texture_extent,
+            array_layer_count: 1,
+            mip_level_count: 1,
+            sample_count,
+            dimension: wgpu::TextureDimension::D2,
+            format: wgpu::TextureFormat::Bgra8Unorm,
+            usage: wgpu::TextureUsage::OUTPUT_ATTACHMENT,
+        };
+
+        let multisampled_frame = device.create_texture(multisampled_frame_descriptor);
+        let mut encoder =
+            device.create_command_encoder(&wgpu::CommandEncoderDescriptor { todo: 0 });
+        {
+            let mut rpass = encoder.begin_render_pass(&wgpu::RenderPassDescriptor {
+                color_attachments: &[wgpu::RenderPassColorAttachmentDescriptor {
+                    attachment: &multisampled_frame.create_default_view(),
```

Comment:
Should the texture and view be created every frame or only when the window size changes?

Reply:
It is a good question, and it's currently unclear because Google is pushing for caching all the things by the implementation, i.e. the device would remember if there is already a good view for a given texture and return it again.
I'd prefer if we didn't rely on this hypothetical behavior :)